src/PlayerbotAI.cpp

```diff
@@ -4124,62 +4142,166 @@ bool PlayerbotAI::AllowActive(ActivityType activityType)
             isLFG = true;
         }
     }
-
     if (sLFGMgr->GetState(bot->GetGUID()) != lfg::LFG_STATE_NONE)
     {
         isLFG = true;
     }
-
     if (isLFG)
-        return true;
-
-    if (activityType != OUT_OF_PARTY_ACTIVITY && activityType != PACKET_ACTIVITY)  // Is in combat. Defend yourself.
-        if (bot->IsInCombat())
-            return true;
+        return ActivePiorityType::IN_LFG;
 
-    if (HasPlayerNearby(300.f))  // Player is near. Always active.
-        return true;
+    // If has real players - slow down continents without player
+    // This means we first disable bots in a different continent/area.
+    if (sRandomPlayerbotMgr->GetPlayers().empty())
+        return ActivePiorityType::IN_EMPTY_SERVER;
 
     // friends always active
-
-    // HasFriend sometimes cause crash, disable
-    // for (auto& player : sRandomPlayerbotMgr->GetPlayers())
-    // {
-    //     if (!player || !player->IsInWorld())
-    //         continue;
+    //  for (auto& player : sRandomPlayerbotMgr->GetPlayers())
+    //  {
+    //      if (!player || !player->IsInWorld())
+    //          continue;
 
     //     if (player->GetSocial()->HasFriend(bot->GetGUID()))
-    //         return true;
+    //         return ActivePiorityType::PLAYER_FRIEND;
     // }
 
-    if (activityType == OUT_OF_PARTY_ACTIVITY ||
-        activityType == GRIND_ACTIVITY)  // Many bots nearby. Do not do heavy area checks.
+    // real guild always active if member+
+    if (IsInRealGuild())
+        return ActivePiorityType::PLAYER_GUILD;
+
+    if (bot->IsBeingTeleported() || !bot->IsInWorld() || !HasRealPlayers(bot->GetMap()))
+        return ActivePiorityType::IN_INACTIVE_MAP;
+
+    // IN_ACTIVE_AREA
+    if (activityType == OUT_OF_PARTY_ACTIVITY || activityType == GRIND_ACTIVITY)
+    {
+        // Many bots nearby. Do not do heavy area checks.
         if (HasManyPlayersNearby())
-            return false;
+            ActivePiorityType::IN_ACTIVE_AREA;
+    }
+}
+
+// Returns the lower and upper bracket for bots to be active.
+// Ie. 10,20 means all bots in this bracket will be inactive below 10% activityMod, all bots in this bracket will be
+// active above 20% activityMod and scale between those values.
+std::pair<uint32, uint32> PlayerbotAI::GetPriorityBracket(ActivePiorityType type)
```

Comment:
The idea of ​​setting multiple active priorities is a good one, I'm just a little worried that it may make the server overhead unpredictable and make debugging more difficult. I'm not sure. We can merge it and see the effect to decide whether to keep it.

Reply:
lets test abit more